File: SlidingWindow/findLongestSubstring.py
def findLongestSubstring(str):
    left = 0
    maximum = 0
    seen = set()
    for i in range(len(str)):
        while(str[i] in seen):
            seen.remove(str[left])
            left = left+1
        seen.add(str[i])
        maximum = max(maximum, i - left +1)
    return maximum



print(findLongestSubstring('rithmschool'));

File: SlidingWindow/findLongestSubstring.js
function findLongestSubstring(str) {
  let left = 0;
  let maximum  = -Infinity;
  let seen = new Set();
  if (str.trim().length === 0) return 0;
  for (let right = 0; right < str.length; right++) {
    while(seen.has(str[right])) {
      seen.delete(str[left]);
      left++;
    }
    seen.add(str[right]);
    maximum = Math.max(maximum, right - left + 1);
  }
  return maximum;
}



console.log(findLongestSubstring('rithmschool'));